### ISSUE TITLE
论文请教

### ISSUE THREAD

**Issue by wangwangwangLi** [Hxxxz0/Free-T2m#2]
您好。拜读了您的大作和代码，有些疑问想请教一下：

1. 公式（1）推导到公式（3），应该是+号？r = \delta u + k^2 u **+** f ? （论文里公式（3）是 r = \delta u + k^2 u **-** f ）
2. 公式（3）中，代码里k(i,j)在建筑物里面取值为0.5，建筑物外面取值为0.2，有什么依据吗？是否可以调整 k(i,j) 的值使得网络性能提升？
3. 公式（5）显示需要计算u(i,j) - u_BC(i,j) 误差的平方，代码里只计算了 u(i,j)的平方，是否有问题？
4. 代码里公式（7）的pathloss 是用numpy计算的，而不是torch计算的。numpy计算的结果在 loss.backward()时好像不会更新，不知道是否有问题。

期待您的回复，谢谢。

**Comment by Hxxxz0** [comment]
符号的这个问题和数据集的正负相关k值的选择是我们在数据集上测试得出来的自己右结果其次，我们现在的计算方式并不是严格的物理损失，只能说是一种类物理的正则化，我们现在也在探求如何去做成严格意义的pinn，如果有兴趣可以一起研究一下。numpy虽然没有计算梯度，但是也是会影响到模型训练的。发自我的 iPhone在 2025年4月26日，15:33，wangwangwangLi ***@***.***> 写道：﻿
wangwangwangLi created an issue (Hxxxz0/Free-T2m#2)
您好。拜读了您的大作和代码，有些疑问想请教一下：

公式（1）推导到公式（3），应该是+号？r = \delta u + k^2 u + f ? （论文里公式（3）是 r = \delta u + k^2 u - f ）
公式（3）中，代码里k(i,j)在建筑物里面取值为0.5，建筑物外面取值为0.2，有什么依据吗？是否可以调整 k(i,j) 的值使得网络性能提升？
公式（5）显示需要计算u(i,j) - u_BC(i,j) 误差的平方，代码里只计算了 u(i,j)的平方，是否有问题？
代码里公式（7）的pathloss 是用numpy计算的，而不是torch计算的。numpy计算的结果在 loss.backward()时好像不会更新，不知道是否有问题。

期待您的回复，谢谢。
